build_scripts/travis_script.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+"""A simple script to run Travis build steps."""
+
+from __future__ import print_function
+
+import collections
+import os
+import sys
+
+import build_utils
+
+def _ismod(f):
+  return f.endswith(".py") and not f.endswith("_test.py")
+
+
+STEP = collections.namedtuple("STEP", ["name", "command"])
+
+PYTYPE_COMMAND = os.path.join("out", "bin", "pytype")
+PYC_DIR = os.path.join(build_utils.PYTYPE_SRC_ROOT, "pytype", "pyc")
+PYC_FILES = [os.path.join(PYC_DIR, f) for f in os.listdir(PYC_DIR) if _ismod(f)]
+TYPECHECK_FILES = [
```

Comment:
The information about which files to type-check is only living here temporarily until we add the functionality that will allow us to put it in setup.cfg, so I think it would be better to keep hard-coding `pytype/compat.py pytype/debug.py pytype/utils.py pytype/pyc --exclude pytype/pyc/*_test.py` in the pytype command rather than trying to turn it into something more maintainable long-term: the former is easier to read, and this won't be around in the long-term anyway.

Reply:
I prefer to keep "--exclude pytype/pyc/*_test.py" as well. However, it was working earlier because the shell does the * expansion. Without the help from shell, this is what I could come up with.

Reply:
see if passing shell=True to run_cmd fixes it

Reply:
That seems reasonable to me.

Reply:
i wonder if we should add --exclude_pattern to pytype, or if that would just complicate the interface

Reply:
There's already an --exclude, but it's currently a command-line-only argument. It's on my TODO list to add it to the file config so that we can move exclude patterns to setup.cfg.

Reply:
--exclude takes files and directories as args, and counts on the shell for globbing; exclude_pattern would take a literal glob pattern and filter out files within analyze_project. potentially a bad idea because you'd have to remember to protect against shell expansion when you pass it in, but as a setup.cfg entry it would work fine.

Reply:
Oh, I see. Rather than having both an --exclude and an --exclude_pattern, I think a more user-friendly thing to do would be to treat an exclude= entry in setup.cfg as a glob pattern under the hood, but I suppose what we end up doing will also depend on what's feasible implementation-wise.

Reply:
the issue is for things like `--exclude=*_test.py`, do we want to exclude `**/*_test.py` or just `*_test.py` files in the current directory? the former is probably more intuitive, but then it will do the wrong thing if bash expands the `*`. (it's a common problem i run into when running commands like `find . -name *.py`)

Reply:
Ah, I see what you mean now. Not sure whether it's better to avoid proliferation of config options or to avoid making the user tediously list a bunch of similar excludes, but I've updated the bug to take this into account.